pkg/provider-local/controller/worker/machines.go

```diff
@@ -145,13 +146,18 @@ func (w *workerDelegate) generateMachineConfig(ctx context.Context) error {
 			ProviderSpec: runtime.RawExtension{Raw: providerConfigBytes},
 		})
 
+		if pool.Priority == nil {
+			pool.Priority = pointer.Int32(0)
+		}
+
 		machineDeployments = append(machineDeployments, worker.MachineDeployment{
 			Name:                         deploymentName,
 			ClassName:                    className,
 			SecretName:                   className,
 			Minimum:                      pool.Minimum,
 			Maximum:                      pool.Maximum,
 			MaxSurge:                     pool.MaxSurge,
+			Priority:                     *pool.Priority,
```

Comment:
```suggestion
			Priority:                     ptr.Deref(pool.Priority, 0),
```

And you could remove the `if` check from line 149.